### ISSUE TITLE
Deleting Folders

### ISSUE THREAD

**Issue by kikoking** [morassman/atom-commander#54]
Hey, I love Atom Commander!! Thank you so much for making it! 8) I was wondering if you could make it so that a folder doesn't have to be empty to delete it? Like when a folder is deleted it will delete everything inside automatically.

**Comment by morassman** [comment]
Thanks @kikoking 

This should work on the local file system, so I assume you are talking about deleting folders over FTP? If it doesn't work locally then it's probably a bug. For FTP, though, it unfortunately doesn't work like that out of the box. I'm not sure whether it's the FTP protocol or the implementation that determines whether a folder must be empty before it can be deleted. I use a third party library for FTP and it requires the folder to be empty.

I can implement this behaviour on top of the library though. It's simply a matter of recursively deleting everything in a folder from the bottom up. I'll look into, maybe not for the next update, but perhaps for the one thereafter.

**Comment by kikoking** [comment]
Yes I meant FTP, and thank you so much! I hope it can be figured out, but commander is still epic either way!! Thanks man!

**Comment by morassman** [comment]
Sure. You're welcome :)

**Comment by Aleksandern** [comment]
I can't delete not empty folder from a remote server (SSH).
> Make sure the folder is empty before deleting it.

Is it possible to fix it?

**Comment by morassman** [comment]
This isn't really a bug, so it doesn't require a fix so to speak. I'm not sure, but AFAIK SFTP servers do not support deleting non-empty folders, but it's up to the client to recursively delete folders. The fact that you cannot do this means that the third party library I use doesn't support it. It's therefore something that I'll have to implement on my side.

**Comment by kikoking** [comment]
@Aleksandern There should be a file manager on your hosting control panel somewhere where you can access your server files and be able to delete them recursively that way, this is what I normally do. Another alternative, which I sometimes use but not often, is using the linux command to delete files in folders recursively via ssh in the command line. This is a very dangerous way of doing it though as you can accidentally delete all of the files on your server by adding the slash to a command like rm -f /*. There is also an ftp code editor out there called coda that can do exactly what you are asking. It can delete a folder recursively down to the very last file that is in it straight from the editor, however it is a paid program, though its only a one time $100 payment. Recursively deleting folders is slow though, as I imagine it would be on any ftp editor that could do it. Another downside for Coda is though that it is only for mac right now too. My last company used coda, it is very easy to use and powerful. I still love to use Atom though, I just love the look and feel of it. I also use it because its free, and because I am a teacher its a lot easier to convince students to use a free text editor, (I find Atom is the best free one) than a paid text editor for programming.
